ONSdigital/ssdc-rm-caseprocessor
# Upgrading to spring boot 3.2.2

# Motivation and Context

Spring Boot 3 has been out for a while and Dependabot wants us to upgrade to it. There's quite a few changes which you look up for yourselves. I listed main changes to each repo on the ticket.

# What has changed



- Move to Jakarta resources instead of Javax
- updated postgresdialect to `PostgreSQLDialect`
- Retry Advice can't be used as a retry listener anymore so I've created one to be used for methods annotated with `@retry`
- ListenableFuture is deprecated in favour of CompletableFuture
- localwebserver comes from the test location now instead of server
# How to test?



- Run `make build` on the exception manager branch and UAC QID service repos
- Go to the ddl repo and run `UAC_QID_SERVICE_BRANCH=1288-upgrade-spring-boot-version-in-java-repos EXCEPTION_MANAGER_BRANCH=1288-upgrade-spring-boot-version-in-java-repos make dev-build
`. This should build the expected ddl changes for you.
- run `make build` on the rest of the java services
- Run the ATs and make sure everything works as expected.
# Links



[Trello](https://trello.com/c/Oo7EZ4q4/)
# Screenshots (if appropriate):
